test/apps/iodemo/io_demo.cc

```diff
@@ -349,48 +349,49 @@ class IoDemoRandom {
     }
 
 private:
+    typedef struct {
+        uint32_t segment;
+        uint32_t seed;
+    } UCS_S_PACKED fill_data_t;
+
     template<typename T>
-    static inline void fill(unsigned &seed, T *buffer, size_t count)
+    static inline void fill(unsigned seed, T *buffer, size_t count)
     {
-        for (size_t i = 0; i < count; ++i) {
-            buffer[i] = i;
+        if (seed == 0) {
+            for (size_t i = 0; i < count; ++i) {
+                buffer[i] = i;
+            }
+            return;
```

Comment:
sorry, i misread the code, didn't notice it's actually filling uint8_t in this case. so it better be a separate function.